addons/mail/wizard/mail_compose_message.py

```diff
@@ -901,8 +907,19 @@ def _process_mail_values_state(self, mail_values_dict):
             elif not mail_to_normalized or not email_re.findall(mail_to):
                 mail_values['state'] = 'cancel'
                 mail_values['failure_type'] = 'mail_email_invalid'
-            elif done_emails is not None and not mailing_document_based:
-                done_emails.append(mail_to)
+            elif mail_to in sent_emails.keys():
+                # lazy eval avoids checking more than necessary
+                was_sent = lambda compared_mail_values: are_same_mail(mail_values, compared_mail_values)
+                was_ever_sent = map(was_sent, sent_emails[mail_to])
+                if any(was_ever_sent):
```

Comment:
Somehow I feel like there is some over engineering here :D can't we simply say
```
mail_to_potential_dupes = sent_emails.get(mail_to) or []
was_sent = any(
    (mail_values['subject'] == sent_mail['subject'] and
     mail_values['body'] == sent_mail['body'])
    for sent_mail in mail_to_potential_dupes
)
```
?
As the 'and' will stop the at first check (subject, easy); anyway comparing bodies is generally straightforward (begins by comparing string sizes, generally they will be different if the content is really dynamic).

Not sure we should compare attachments... if subject and body is the same, well ... maybe. Or not. Mmmmh.

Reply:
Comparison with the attachments is to ensure we don't mark an invoice as 'sent' just because the subject and message were the same. I think in that case we'd rather just send duplicates, just in case.

Otherwise, yes indeed it was very overengineered ^^
I guess `map` really isn't useful 99% of the time when you can just have iterators without using it